### ISSUE TITLE
'BuildContext's across async gaps false positive?

### ISSUE THREAD

**Issue by stephane-archer** [dart-lang/linter#4955]
```
                      if (context.mounted) {
                        ScaffoldMessenger.of(context).showSnackBar(snackBar);
                      }
``` 
ok but:
``` 
                      if (context.mounted == false) {
                        return;
                      }
                      ScaffoldMessenger.of(context).showSnackBar(snackBar);
``` 

`Don't use 'BuildContext's across async gaps.
Try rewriting the code to not use the 'BuildContext', or guard the use with a 'mounted' check.`

**Comment by srawlins** [comment]
A possible gap in the rule! I think this doesn't come up because it is more idiomatic to write `if (!context.mounted)`.

**Comment by srawlins** [comment]
Actually I cannot reproduce this at HEAD. Can you say what version of Dart/Flutter you are using, and provide a more full example (just the full function body there should be sufficient).

**Comment by srawlins** [comment]
Oops, apologies. I can reproduce. :D

**Comment by stephane-archer** [comment]
@srawlins no worries
The idiomatic way would fail silently or need an else close to handle the error which is quite verbose
I'm not sure why it's the idiomatic way :)
```
if (context.mounted == false) {
                        throw "I don't fail silently!";
 }
ScaffoldMessenger.of(context).showSnackBar(snackBar);
```

**Comment by srawlins** [comment]
Hmm, I might have not been clear. I mean that the idiomatic way to write `context.mounted == false` is `!context.mounted`.

**Comment by stephane-archer** [comment]
@srawlins ho I see, I misunderstood what you meant, sorry. 
I hope to allow `context.mounted == false` would be easy to add.

**Comment by srawlins** [comment]
Fixed with https://github.com/dart-lang/sdk/commit/d728ec3c65aeafe4c1559e1875c5f09e1e6573a9

**Comment by stephane-archer** [comment]
@srawlins you are the best 🤯
